Tests/SwiftDocCTests/Indexing/NavigatorIndexTests.swift

```diff
@@ -894,6 +894,32 @@ Root
         )
     }
     
+    /// Test the curation of a symbol in a page where the page language is different from the symbol language.
+    /// This occurs in multi-language frameworks when a symbol in one language is referenced in another, e.g.
+    /// a Swift-only symbol includes a link to an Objective-C only symbol.
+    func testCurateSymbolsInPageWithNoCommonLanguage() async throws {
+        let navigatorIndex = try await generatedNavigatorIndex(
+            for: "MixedLanguageFrameworkSingleLanguageCuration",
+            bundleIdentifier: "org.swift.mixedlanguageframework"
+        )
+
+        XCTAssertEqual(
+            navigatorIndex.navigatorTree.root.children
+                .first { $0.item.title == "Swift" }?
+                .children
+                .first { $0.item.title == "MixedLanguageFramework" }?
+                .children
+                .first { $0.item.title == "SwiftOnlyStruct2" }?
+                .children
+                .contains { $0.item.title == "ObjectiveCOnlyClass" },
```

Comment:
This behavior doesn't seem correct to me. 

If I was browsing the Swift version of this documentation I would expect to not see any symbols that don't have Swift representations. 

Similarly, if I was browsing the Objective-C version of this documentation I would expect to not see any symbols that don't have an Objective-C representation. Curating a symbol under a page that it has no overlapping languages with, like this, feels like something we should raise a warning about if we don't already. 

IMO if a symbol is _only_ curated in a location that is has no overlapping languages with, we should treat it as uncrated and fall back to the automatic default curation for that symbol.